### DISCUSSION TITLE
Network outage, use startPlayer to play URL, error callback required

### DISCUSSION THREAD

**Discussion by liaojh554** [dooboolab/react-native-audio-recorder-player#483] [General]
> Please fill the template to help you out. Also, please try the `Example` project compare before submiting the issue when you have certain issue with your project setup.

### Version of react-native-audio-recorder-player
 - react-native-audio-recorder-player:3.3.0

### Version of React Native
 - React Native: 0.63.4

### Platforms you faced the error (IOS or Android or both?)
 - both
### Expected behavior
 - Network outage, use startPlayer to play URL, error callback required
### Actual behavior
 - Android and ios have no error callbacks
### Steps to reproduce the behabior
 1. Shut down the network
 2. use startPlayer to play URL

**Comment by nastarfan** [comment]
hi @hyochan , do you have thoughts on this?
This library works great, but when playing track from the network, [it loads a bit slow](https://github.com/hyochan/react-native-audio-recorder-player/issues/437) and when the network is really bad, it would make the app freeze (i guess it's blocking the js). This particular issue is really painful, other things work great!

**Comment by hyochan** [comment]
Looks like this is a great issue to be covered. I've tried to handle the error message when URL is not player in #480, however, they do not throw the error 🤔

What would be the good proposal to handle such issue like this?

**Reply by MinaSamir11** [reply]
please note that while waiting for the callback error application got freeze till the callback returned. 
[ANDROID]

**Reply by MinaSamir11** [reply]
It gives an error after freezing the app
[Error: Prepare failed.: status=0x1] 

I See in the android native module 
you are doing 
  /**
      * Set timer task to send an event to RN.
*/
                 
  may because of this timer freeze the app. !!

My suggestion is to put it inside a thread.
